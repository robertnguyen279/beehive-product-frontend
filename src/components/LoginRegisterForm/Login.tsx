import React from 'react';
import { Input, CheckBox } from 'components/Forms';
import UserIcon from 'assets/icons/user.svg';
import PasswordIcon from 'assets/icons/key.svg';
import Button from 'components/Button';
import { useTranslation } from 'react-i18next';
import { useFormik } from 'formik';
import * as Yup from 'yup';
import { LoginRegisterForm } from 'types';
import GoogleLogin from 'react-google-login';
import GoogleIcon from 'assets/icons/google.svg';
import FacebookIcon from 'assets/icons/facebook.svg';
// @ts-ignore
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import { loginUser, loginByGoogle, loginByFacebook } from 'actions/users';
import { useDispatch } from 'react-redux';

const Login = ({ onChange }: LoginRegisterForm): React.ReactElement => {
  const { t } = useTranslation(['Common']);
  const dispatch = useDispatch();
  const formik = useFormik({
    initialValues: {
      email: '',
      password: '',
    },
    validationSchema: Yup.object({
      email: Yup.string().email(t('Common:form-email-error')).required(t('Common:form-required')),
      password: Yup.string().required(t('Common:form-required')),
    }),
    // @ts-ignore
    onSubmit: ({ email, password, remember }) => {
      dispatch(loginUser({ email, password, remember }));
    },
  });

  const responseGoogle = (res: any): void => {
    const { familyName, givenName, imageUrl, email } = res.profileObj;
    dispatch(
      loginByGoogle({ firstName: familyName, lastName: givenName, avatar: imageUrl, email, token: res.tokenId }),
    );
  };

  const responseFacebook = (res: any): void => {
    const { name, email } = res;
    dispatch(loginByFacebook({ name, email, avatar: res.picture.data.url }));
  };

  return (
    <form
      className="w-full flex flex-col items-center justify-center animate__animated animate__fadeInRight"
      method="POST"
      onSubmit={formik.handleSubmit}
    >
      <Input
        value={formik.values.email}
        onChange={formik.handleChange}
        type="text"
        placeholder={t('Common:form-email-placeholder')}
        name="email"
        className="mb-3 w-4/6"
        prefixIcon={UserIcon}
        error={formik.errors.email && formik.touched.email ? formik.errors.email : false}
      />
      <Input
        value={formik.values.password}
        onChange={formik.handleChange}
        type="password"
        placeholder={t('Common:form-password-placeholder')}
        name="password"
        className="mb-3 w-4/6"
        prefixIcon={PasswordIcon}
        error={formik.errors.password && formik.touched.password ? formik.errors.password : false}
      />
      <div className="flex justify-between items-center w-4/6 pb-3">
        <CheckBox
          name="remember"
          value="remember"
          label={t('Common:checkbox-remember')}
          onChange={formik.handleChange}
        />
        <div className="hover-primary text-sm text-gray-600 cursor-pointer">{t('Common:form-password-lost')}</div>
      </div>
      <Button className="w-4/6 mb-2" type="submit">
        {t('Common:button-login')}
      </Button>
      <div className="flex w-4/6 justify-center items-center py-3">
        <GoogleLogin
          clientId="22234867800-4shnc1g407oks72lciep89ht3kdaeqfn.apps.googleusercontent.com"
          buttonText="Login"
          onSuccess={responseGoogle}
          onFailure={responseGoogle}
          cookiePolicy={'single_host_origin'}
          render={(renderProps) => (
            <img
              src={GoogleIcon}
              alt="google-auth"
              className="w-8 h-8 cursor-pointer mx-5"
              onClick={renderProps.onClick}
            />
          )}
        />
        <FacebookLogin
          appId="441079470350681"
          autoLoad={false}
          callback={responseFacebook}
          fields="name,email,picture"
          // @ts-ignore
          render={(renderProps) => (
            <img
              src={FacebookIcon}
              alt="google-auth"
              className="w-8 h-8 cursor-pointer mx-5"
              onClick={renderProps.onClick}
            />
          )}
        />
      </div>
      <div className="text-sm text-primary cursor-pointer" onClick={() => onChange()}>
        {t('Common:form-or-sign-up')}
      </div>
    </form>
  );
};

export default Login;
