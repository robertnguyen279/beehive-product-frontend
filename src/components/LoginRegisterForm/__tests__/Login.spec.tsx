import React from 'react';
import { shallow } from 'enzyme';

import Login from '../Login';
// const clickFn = jest.fn();

describe('Login', () => {
  const component = shallow(
    <Login
      onChange={() => {
        console.log('change');
      }}
    />,
  );

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
