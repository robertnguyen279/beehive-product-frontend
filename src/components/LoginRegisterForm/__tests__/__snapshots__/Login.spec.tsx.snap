// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<form
  className="w-full flex flex-col items-center justify-center animate__animated animate__fadeInRight"
  method="POST"
  onSubmit={[Function]}
>
  <Input
    className="mb-3 w-4/6"
    error={false}
    name="email"
    onChange={[Function]}
    placeholder="Common:form-email-placeholder"
    prefixIcon="test-file-stub"
    type="text"
    value=""
  />
  <Input
    className="mb-3 w-4/6"
    error={false}
    name="password"
    onChange={[Function]}
    placeholder="Common:form-password-placeholder"
    prefixIcon="test-file-stub"
    type="password"
    value=""
  />
  <div
    className="flex justify-between items-center w-4/6 pb-3"
  >
    <CheckBox
      label="Common:checkbox-remember"
      name="Remember"
      onChange={[Function]}
      value="Remember"
    />
    <div
      className="hover-primary text-sm text-gray-600 cursor-pointer"
    >
      Common:form-password-lost
    </div>
  </div>
  <Button
    className="w-4/6 mb-2"
    type="submit"
  >
    Common:button-login
  </Button>
  <div
    className="flex w-4/6 justify-center items-center py-3"
  >
    <y
      accessType="online"
      buttonText="Login"
      clientId="22234867800-4shnc1g407oks72lciep89ht3kdaeqfn.apps.googleusercontent.com"
      cookiePolicy="single_host_origin"
      disabledStyle={
        Object {
          "opacity": 0.6,
        }
      }
      fetchBasicProfile={true}
      icon={true}
      isSignedIn={false}
      onFailure={[Function]}
      onRequest={[Function]}
      onSuccess={[Function]}
      prompt=""
      render={[Function]}
      scope="profile email"
      tag="button"
      theme="light"
      type="button"
      uxMode="popup"
    />
    <t
      appId="441079470350681"
      authType=""
      autoLoad={false}
      callback={[Function]}
      cookie={false}
      disableMobileRedirect={false}
      fields="name"
      isMobile={false}
      language="en_US"
      onFailure={null}
      redirectUri="http://localhost/"
      render={[Function]}
      responseType="code"
      returnScopes={false}
      scope="public_profile,email"
      state="facebookdirect"
      version="2.3"
      xfbml={false}
    />
  </div>
  <div
    className="text-sm text-primary cursor-pointer"
    onClick={[Function]}
  >
    Common:form-or-sign-up
  </div>
</form>
`;
